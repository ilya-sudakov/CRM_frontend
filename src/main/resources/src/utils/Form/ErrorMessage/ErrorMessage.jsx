import React, { useEffect, useCallback } from "react";
import PropTypes from "prop-types";
import "./ErrorMessage.scss";

const ErrorMessage = ({ showError = false, setShowError, message = "" }) => {
  const clickOnErrorWindow = (event) => {
    event.preventDefault();
    const classList = event.target.classList;
    if (
      !(classList[0] === "window_error") &&
      !classList.contains("window_error__exit") &&
      !classList.contains("window_error__bar") &&
      !classList.contains("main-window__button")
    ) {
      return setShowError(true);
    }

    return setShowError(false);
  };

  const handleCloseWindow = useCallback((event) => {
    if (event.key === "Enter" || event.key === "Escape") {
      event.preventDefault();
      setShowError(false);
    }
  }, []);

  useEffect(() => {
    document.addEventListener("keydown", handleCloseWindow, false);
    return () => {
      document.removeEventListener("keydown", handleCloseWindow, false);
    };
  }, []);

  return (
    <div
      className={
        showError ? "window_error" : "window_error window_error--hidden"
      }
      onClick={clickOnErrorWindow}
    >
      <div
        className={
          showError
            ? "window_error__content"
            : "window_error__content window_error__content--hidden"
        }
      >
        <div className="window_error__title">
          Ошибка!
          <div className="window_error__exit" onClick={clickOnErrorWindow}>
            <div
              className="window_error__bar"
              onClick={clickOnErrorWindow}
            ></div>
            <div
              className="window_error__bar"
              onClick={clickOnErrorWindow}
            ></div>
          </div>
        </div>
        <div className="window_error__message">{message}</div>
        <div className="main-window__button" onClick={clickOnErrorWindow}>
          ОК
        </div>
      </div>
    </div>
  );
};

export default ErrorMessage;

ErrorMessage.propTypes = {
  showError: PropTypes.bool,
  setShowError: PropTypes.func,
  message: PropTypes.string,
};
