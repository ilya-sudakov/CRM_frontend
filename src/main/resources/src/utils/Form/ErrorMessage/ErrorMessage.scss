@import "../../../variables.scss";

.window_error {
  display: flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  width: 100vw;
  height: 100vh;
  background-color: rgba(#000000, 0.3);
  top: 0;
  left: 0;
  z-index: 999;
  transition: 200ms ease-in-out;

  &--hidden {
    background-color: rgba(#000000, 0);
    transform: scale(1);
    visibility: hidden;
  }

  &__content {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: flex-start;
    width: 300px;
    min-height: 150px;
    background-color: #eeeeee;
    border: #cccccc;
    border-radius: 3px;
    border: 1px solid #999999;
    padding: 15px;
    overflow-y: auto;
    position: relative;
    transform: scale(1);
    transition: 100ms ease-in-out;

    &--hidden {
      overflow: hidden;
      transform: scale(0);
    }

    .tableview_products {
      width: 100%;
    }

    .window_error__title {
      text-align: center;
      padding: 0px 0px;
      margin-bottom: 5px;
      font-size: 125%;
      display: flex;
      width: 100%;
      justify-content: center;
      align-items: center;
      margin-bottom: 10px;
      position: relative;
      color: #945555;
    }

    .window_error__message {
      font-size: 80%;
      margin-bottom: 15px;
    }

    .main-window__button {
      padding: 10px 15px;
      margin: 0 auto;
      margin-top: auto;
      width: fit-content;
      // margin-top: 15px;
      min-width: 100px;
    }

    .window_error__exit {
      position: absolute;
      width: 45px;
      height: 45px;
      right: -5px;
      border-radius: 5px;
      transition: 0.2s ease-in-out;
      cursor: pointer;

      &:hover {
        background-color: #cccccc;
      }
      .window_error__bar {
        height: 2px;
        width: 25px;
        content: "";
        background-color: #777777;
        position: absolute;
        right: 9px;
        bottom: 21px;

        &:first-of-type {
          transform: rotate(45deg);
        }
        &:last-of-type {
          transform: rotate(-45deg);
        }
      }
    }

    .searchbar {
      width: 100%;
    }
  }

  @media (max-width: 768px) {
    .window_error__content {
      width: 250px;
    }
  }
}
