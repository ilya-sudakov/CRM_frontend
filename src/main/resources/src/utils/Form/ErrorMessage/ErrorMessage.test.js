import React from "react";
import ErrorMessage from "./ErrorMessage.jsx";
import { render, cleanup, screen, fireEvent } from "@testing-library/react";
// import { renderWithRouter } from "../../../utils/testing/functions.js";
import "@testing-library/jest-dom/extend-expect";

describe("ErrorMessage component", () => {
  afterEach(cleanup);

  it("renders", () => {
    render(<ErrorMessage />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<ErrorMessage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("closes window", () => {
    let showError = true;
    render(
      <ErrorMessage
        showError={showError}
        setShowError={(value) => (showError = value)}
      />
    );
    const exitButton = document.getElementsByClassName("window_error__exit")[0];
    const exitBar = document.getElementsByClassName("window_error__bar")[0];
    const button = document.getElementsByClassName("main-window__button")[0];
    fireEvent.click(exitButton);
    expect(showError).toBe(false);
    fireEvent.click(exitBar);
    fireEvent.click(button);
  });
});
