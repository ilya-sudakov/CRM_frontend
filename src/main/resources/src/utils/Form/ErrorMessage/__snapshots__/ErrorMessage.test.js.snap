// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="window_error window_error--hidden"
  >
    <div
      class="window_error__content window_error__content--hidden"
    >
      <div
        class="window_error__title"
      >
        Ошибка!
        <div
          class="window_error__exit"
        >
          <div
            class="window_error__bar"
          />
          <div
            class="window_error__bar"
          />
        </div>
      </div>
      <div
        class="window_error__message"
      />
      <div
        class="main-window__button"
      >
        ОК
      </div>
    </div>
  </div>
</DocumentFragment>
`;
